### DISCUSSION TITLE
Get class with decorator and implements

### DISCUSSION THREAD

**Discussion by grosch-intl** [eslint/eslint#17045] [Q&A]
I want to match classes that have an @Component decorator and also implement an interface.  I tried to use the following selector to no avail:

```ts
[`ClassDeclaration > Decorator[expression.callee.name="Component"] + TSClassImplements`](node: TSClassImplements): void {
```

Decorator and TSClassImplements are both direct children of a ClassDeclaration.  Can I write such a selector?  Also, is it possible to get multiple items passed into the method?  Right now the node seems to be the last thing that matches.  I'd like to get both the Decorator and the TSClassImplements, for example.

**Comment by mdjermanovic** [comment]
`+` can only match nodes in the same collection. For example, it can match function parameter's sibling function parameter as they're both in the same `.params[]` collection. It doesn't match other direct children of the same parent.

You could try this selector:

```
ClassDeclaration[implements] > Decorator[expression.callee.name="Component"]
```

This matches the decorator if its parent class declaration has `implements`. Your method will get the decorator node, and from there you can access the TSClassImplements node by `node.parent.implements`.

> Also, is it possible to get multiple items passed into the method?

No, selectors always match one node and that node is passed into the method.

**Reply by grosch-intl** [reply]
If I use your selector then when I take `node.parent.implements` I've got an array of implementations, instead of just the one.  Is it OK to just loop through them, possibly reporting an issue for each one?

**Reply by mdjermanovic** [reply]
Yes, I missed the fact that `implements` is a collection.

In that case, it seems better to also check the length:

```
ClassDeclaration[implements.length>0] > Decorator[expression.callee.name="Component"]
```

> Is it OK to just loop through them, possibly reporting an issue for each one?

Sounds good to me.
